File: app/src/main/java/mishka/rssreader/data/model/EnclosureListConverter.java
package mishka.rssreader.data.model;

import android.arch.persistence.room.TypeConverter;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import java.lang.reflect.Type;
import java.util.List;

import mishka.rssreader.data.model.RssItem;

public class EnclosureListConverter {

    @TypeConverter
    public List<RssItem.Enclosure> jsonToEnclosure(String json) {
        Type type = new TypeToken<List<RssItem.Enclosure>>() {}.getType();
        return new Gson().fromJson(json, type);
    }

    @TypeConverter
    public String enclosuresToJson(List<RssItem.Enclosure> enclosures) {
        return new Gson().toJson(enclosures);
    }
}

File: app/src/main/java/mishka/rssreader/data/model/EnclosureListConverter.kt
package mishka.rssreader.data.model

import android.arch.persistence.room.TypeConverter
import com.google.gson.Gson

import com.google.gson.reflect.TypeToken

class EnclosureListConverter {

    @TypeConverter
    fun jsonToEnclosure(json: String): List<RssItem.Enclosure> {
        val type = object : TypeToken<List<RssItem.Enclosure>>() {}.getType()
        return Gson().fromJson(json, type)
    }

    @TypeConverter
    fun enclosuresToJson(enclosures: List<RssItem.Enclosure>): String {
        return Gson().toJson(enclosures)
    }


}
